src/cwrapper.h

```diff
@@ -0,0 +1,98 @@
+#ifndef CWRAPPER_H
+#define CWRAPPER_H
+
+#include <gmp.h>
+
+#ifdef __cplusplus
+#include "basic.h"
+#include "constants.h"
+namespace CSymPy {
+    class CWrapper {
+        public:
+            RCP<const Basic> ptr;
+            CWrapper() {
+                ptr = zero;
+            }
+    };
+}
+using CSymPy::CWrapper;
+
+extern "C" {
+
+#else
+typedef struct CWrapper CWrapper;
+#endif
```

Comment:
I'll move the definition to the cpp file. `extern "C" {}` is not recognized in `C`, so only that will remain of the C++ part as it is needed to avoid name mangling.